packages/neo-one-smart-contract-codegen/src/vue/genVue.ts

```diff
@@ -0,0 +1,75 @@
+import { getCreateSmartContractName } from '../contract';
+import { ContractPaths } from '../type';
+import { getRelativeImport, lowerCaseFirst } from '../utils';
+
+export const genVue = ({
+  contractsPaths,
+  vuePath,
+  commonTypesPath,
+  clientPath,
+}: {
+  readonly contractsPaths: ReadonlyArray<ContractPaths>;
+  readonly vuePath: string;
+  readonly commonTypesPath: string;
+  readonly clientPath: string;
+}) => ({
+  js: `
+import { Client, DeveloperClient, LocalClient } from '@neo-one/client';
+import { createClient, createDeveloperClients, createLocalClients } from '${getRelativeImport(vuePath, clientPath)}';
+
+${contractsPaths
+    .map(
+      ({ name, createContractPath }) =>
+        `import { ${getCreateSmartContractName(name)} } from '${getRelativeImport(vuePath, createContractPath)}';`,
+    )
+    .join('\n')}
+
+export class ContractsService {
+  constructor() {
+    this.client = createClient();
+    this.developerClients = createDeveloperClients();
+    this.localClients = createLocalClients();
+    ${contractsPaths
+      .map(({ name }) => `this.${lowerCaseFirst(name)} = ${getCreateSmartContractName(name)}(this.client);`)
+      .join('\n    ')}
+  }
+}
+    `,
+  ts: `
+import { Client, DeveloperClient, LocalClient } from '@neo-one/client';
+import { createClient, createDeveloperClients, createLocalClients } from '${getRelativeImport(vuePath, clientPath)}';
+import { Contracts } from '${getRelativeImport(vuePath, commonTypesPath)}';
+
+${contractsPaths
+    .map(
+      ({ name, createContractPath }) =>
+        `import { ${getCreateSmartContractName(name)} } from '${getRelativeImport(vuePath, createContractPath)}';`,
+    )
+    .join('\n')}
+
+export interface DeveloperClients {
+  readonly [network: string]: DeveloperClient;
+};
+export interface LocalClients {
+  readonly [network: string]: LocalClient;
+};
+
+export class ContractsService {
```

Comment:
Export this as a singleton. I.e. this should have `export const contractsService = new ContractsService()`.